pylint/reporters/text.py

```diff
@@ -193,18 +204,37 @@ class ColorizedTextReporter(TextReporter):
     """Simple TextReporter that colorizes text output"""
 
     name = "colorized"
-    COLOR_MAPPING = {
-        "I": ("green", None),
-        "C": (None, "bold"),
-        "R": ("magenta", "bold, italic"),
-        "W": ("magenta", None),
-        "E": ("red", "bold"),
-        "F": ("red", "bold, underline"),
-        "S": ("yellow", "inverse"),  # S stands for module Separator
+    COLOR_MAPPING: ColorMappingDict = {
+        "I": MessageStyle("green", ()),
+        "C": MessageStyle(None, ("bold",)),
+        "R": MessageStyle("magenta", ("bold", "italic")),
+        "W": MessageStyle("magenta", ()),
+        "E": MessageStyle("red", ("bold",)),
+        "F": MessageStyle("red", ("bold", "underline")),
+        "S": MessageStyle("yellow", ("inverse",)),  # S stands for module Separator
     }
 
-    def __init__(self, output=None, color_mapping=None):
+    def __init__(
+        self,
+        output: Optional[TextIO] = None,
+        color_mapping: Union[
+            ColorMappingDict, Dict[str, Tuple[Optional[str], Optional[str]]], None
+        ] = None,
+    ) -> None:
         TextReporter.__init__(self, output)
+        # pylint: disable-next=fixme
+        # TODO: Remove DeprecationWarning and only accept ColorMappingDict as color_mapping parameter
+        if color_mapping and not isinstance(
+            list(color_mapping.values())[0], MessageStyle
+        ):
+            warnings.warn(
+                "In pylint 3.0, the ColoreziedTextReporter will only accept ColorMappingDict as color_mapping parameter",
+                DeprecationWarning,
+            )
+            for key, value in color_mapping.items():
+                color = value[0]
+                style_attrs = tuple(_splitstrip(value[1]))
+                color_mapping[key] = MessageStyle(color, style_attrs)  # type: ignore
```

Comment:
We shouldn't need to add `type: ignore` here. Maybe create a temp dict with type `ColorMappingDict` to assign those to. After the for loop assign the temp one to `color_mapping`.

Reply:
I figured I would add a `type: ignore` for the time being and remove after deprecation, but I will change this!

Reply:
I tried this, but couldn't get it to work.
With:
```python
if color_mapping and not isinstance(
           color_mapping, ColorMappingDict
        ):
            warnings.warn(
                "In pylint 3.0, the ColoreziedTextReporter will only accept ColorMappingDict as color_mapping parameter",
                DeprecationWarning,
            )
            temp_color_mapping = ColorMappingDict()
            for key, value in color_mapping.items():
                color = value[0]
                style_attrs = tuple(_splitstrip(value[1]))
                temp_color_mapping[key] = MessageStyle(color, style_attrs)
            color_mapping = temp_color_mapping
        self.color_mapping = color_mapping or ColorizedTextReporter.COLOR_MAPPING
```
<img width="709" alt="Schermafbeelding 2021-09-20 om 19 57 32" src="https://user-images.githubusercontent.com/13665637/134050913-5c35bc07-b606-4b1c-86e9-1eb545c46cdd.png">

`color_mapping` keeps getting inferred as possibly being `Dict` even though `color_mapping = temp_color_mapping` makes this impossible. I think the `type: ignore` might be better for the time being.